include/oc_export.h

```diff
@@ -0,0 +1,42 @@
+/*
+// Copyright (c) 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#ifndef OC_EXPORT_H
+#define OC_EXPORT_H
+
+#if defined _WIN32 || defined __CYGWIN__ || defined __MINGW32__
+#define OC_EXPORT __declspec(dllexport)
+#define OC_IMPORT __declspec(dllimport)
+#elif __GNUC__ >= 4
+#define OC_EXPORT __attribute__((visibility("default")))
+#define OC_IMPORT
+#else /* __GNUC__ < 4 */
+#warning "Shared libraries not supported"
+#define OC_EXPORT
+#define OC_IMPORT
+#endif /* defined _WIN32 || defined __CYGWIN__ || defined __MINGW32__ */
+
+#ifdef OC_SHARED_LIBRARY
+#ifdef OC_SHARED_LIBRARY_EXPORTS
```

Comment:
Nitpick:
```suggestion
#ifdef OC_SHARED_LIBRARY_EXPORT
```
In my opinion this would be more consistent with `OC_EXPORT`.

Reply:
Yeah, I agree. By default CMake adds `${TARGET_NAME}Exports` compile definition to shared library targets, so I kept the plural when I redefined it, but there is no reason for it.